packages/@aws-cdk/aws-ec2/lib/subnet.ts

```diff
@@ -112,4 +126,53 @@ class ContainsIpAddressesSubnetFilter extends SubnetFilter {
       return cidrBlockObjs.some(cidr => subnetCidrBlock.containsCidr(cidr));
     });
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Chooses subnets based on the subnetId
+ */
+class SubnetIdSubnetFilter extends SubnetFilter {
+
+  private readonly subnetIds: string[];
+
+  constructor(subnetIds: string[]) {
+    super();
+    this.subnetIds = subnetIds;
+  }
+
+  /**
+   * Executes the subnet filtering logic.
+   */
+  public selectSubnets(subnets: ISubnet[]): ISubnet[] {
+    return subnets.filter(subnet => this.subnetIds.includes(subnet.subnetId));
+  }
+}
+
+/**
+ * Chooses subnets based on the CIDR Netmask
+ */
+class CidrMaskSubnetFilter extends SubnetFilter {
+  private readonly mask: number
+
+  constructor(mask: number) {
+    super();
+    if (mask < 16 || mask > 28 ) {
```

Comment:
Outsider comment: this check isn't necessary. The Subnet class should already prevent creation of a subnet that isn't like    28 >= cidr mask  >= 16.

If AWS supports a different range of subnets in the future, this would need to update as well.

Reply:
Agreed. I think this check is redundant.